### PR TITLE
python3Packages.ibis-framework: 5.1.0 -> 6.0.0

### PR DESCRIPTION
**Pull request by cpcloud**
## Description of changes

- ibis changelog: https://ibis-project.org/release_notes/#600-2023-07-05
- sqlglot commit compare: https://github.com/tobymao/sqlglot/compare/v16.3.1...v17.8.6
- clickhouse-connect page: https://clickhouse.com/docs/en/integrations/python

## Things done

~Still debugging a couple of ibis-framework test failures, but otherwise things are working.~

Failing tests are an upstream bug. Deselecting tests for now, will fix upstream.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by cpcloud** [comment]
Result of `nixpkgs-review pr 246170` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>14 packages built:</summary>
  <ul>
    <li>mycli</li>
    <li>mycli.dist</li>
    <li>python310Packages.clickhouse-connect</li>
    <li>python310Packages.clickhouse-connect.dist</li>
    <li>python310Packages.ibis-framework</li>
    <li>python310Packages.ibis-framework.dist</li>
    <li>python310Packages.sqlglot</li>
    <li>python310Packages.sqlglot.dist</li>
    <li>python311Packages.clickhouse-connect</li>
    <li>python311Packages.clickhouse-connect.dist</li>
    <li>python311Packages.ibis-framework</li>
    <li>python311Packages.ibis-framework.dist</li>
    <li>python311Packages.sqlglot</li>
    <li>python311Packages.sqlglot.dist</li>
  </ul>
</details>

**Comment by cpcloud** [comment]
Result of `nixpkgs-review pr 246170` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python310Packages.ibis-framework</li>
    <li>python310Packages.ibis-framework.dist</li>
    <li>python311Packages.ibis-framework</li>
    <li>python311Packages.ibis-framework.dist</li>
  </ul>
</details>
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>mycli</li>
    <li>mycli.dist</li>
    <li>python310Packages.clickhouse-connect</li>
    <li>python310Packages.clickhouse-connect.dist</li>
    <li>python310Packages.sqlglot</li>
    <li>python310Packages.sqlglot.dist</li>
    <li>python311Packages.clickhouse-connect</li>
    <li>python311Packages.clickhouse-connect.dist</li>
    <li>python311Packages.sqlglot</li>
    <li>python311Packages.sqlglot.dist</li>
  </ul>
</details>

**Comment by cpcloud** [comment]
Result of `nixpkgs-review pr 246170` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>14 packages built:</summary>
  <ul>
    <li>mycli</li>
    <li>mycli.dist</li>
    <li>python310Packages.clickhouse-connect</li>
    <li>python310Packages.clickhouse-connect.dist</li>
    <li>python310Packages.ibis-framework</li>
    <li>python310Packages.ibis-framework.dist</li>
    <li>python310Packages.sqlglot</li>
    <li>python310Packages.sqlglot.dist</li>
    <li>python311Packages.clickhouse-connect</li>
    <li>python311Packages.clickhouse-connect.dist</li>
    <li>python311Packages.ibis-framework</li>
    <li>python311Packages.ibis-framework.dist</li>
    <li>python311Packages.sqlglot</li>
    <li>python311Packages.sqlglot.dist</li>
  </ul>
</details>

**Comment by cpcloud** [comment]
@SuperSandro2000 Would you mind reviewing this one?

File: pkgs/development/python-modules/ibis-framework/default.nix [hunk 1]
```diff
@@ -166,17 +145,12 @@ buildPythonPackage rec {
   preCheck = ''
     set -eo pipefail
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/ibis-framework/default.nix:147]:
```suggestion
```
I don't think this is necessary

File: pkgs/development/python-modules/ibis-framework/default.nix [hunk 2]
```diff
@@ -60,99 +57,81 @@
...
-    "--deselect=ibis/backends/tests/test_export.py::test_to_pyarrow_batches_borked_types"
-    "--deselect=ibis/backends/tests/test_export.py::test_table_pyarrow_batch_chunk_size"
-    "--deselect=ibis/backends/tests/test_export.py::test_column_pyarrow_batch_chunk_size"
-    "--deselect=ibis/backends/tests/test_export.py::test_roundtrip_partitioned_parquet"
-    "--deselect=ibis/tests/sql/test_sqlalchemy.py::test_order_by"
-    "--deselect=ibis/tests/sql/test_sqlalchemy.py::test_no_cart_join"
-    "--deselect=ibis/backends/tests/test_temporal.py::test_temporal_binop"
-    "--deselect=ibis/backends/tests/test_temporal.py::test_interval_literal"
+    # fails with pandas <2
+  ] ++ lib.optionals (lib.versionOlder pandas.version "2.0.0") [
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/ibis-framework/default.nix:132]: Do we want to keep this for overrides? We have pandas 2 so we could also drop it.

File: pkgs/development/python-modules/ibis-framework/default.nix [hunk 3]
```diff
@@ -60,99 +57,81 @@
...
-    (fetchpatch {
-      name = "fix-datafusion-compilation.patch";
-      url = "https://github.com/ibis-project/ibis/commit/009230421b2bc1f86591e8b850d37a489e8e4f06.patch";
-      hash = "sha256-5NHkgc8d2bkOMpbY1vme1XgNfyHSr0f7BrR3JTTjjPI=";
-    })
-  ];
-
   nativeBuildInputs = [
     poetry-core
+    poetry-dynamic-versioning
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/ibis-framework/default.nix:88]: That sounds like we maybe need to fix the python package version. Did you check if the version in the installed package is correct?

File: pkgs/development/python-modules/clickhouse-connect/default.nix [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+  nativeCheckInputs = [ pytestCheckHook pytest-dotenv ]
+    ++ passthru.optional-dependencies.sqlalchemy
+    ++ passthru.optional-dependencies.numpy;
+
+  # these tests require a running clickhouse instance
+  pytestFlagsArray = [
+    "--deselect=tests/integration_tests"
+    "--deselect=tests/tls"
+  ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/clickhouse-connect/default.nix:60]: Why not use disabledTestFiles? That also checks if the directory still exists

File: pkgs/development/python-modules/clickhouse-connect/default.nix [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+, pytz
+, zstandard
+, lz4
+  ### extras_require
+, sqlalchemy
+, numpy
+, pandas
+, pyarrow
+, orjson
+  ### not in tests_require, but should be
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/clickhouse-connect/default.nix:21]:
```suggestion
  # build_requires
, cython_3
  # install_requires
, certifi
, importlib-metadata
, urllib3
, pytz
, zstandard
, lz4
  # extras_require
, sqlalchemy
, numpy
, pandas
, pyarrow
, orjson
```

File: pkgs/development/python-modules/ibis-framework/default.nix (hunk 3 above)
Comment (cpcloud) [reply pkgs/development/python-modules/ibis-framework/default.nix:88]: It's correct. This dependency is roughly equivalent to depending on `setuptools-scm`. What's the problem with depending on it?

File: pkgs/development/python-modules/clickhouse-connect/default.nix (hunk 4 above)
Comment (cpcloud) [reply pkgs/development/python-modules/clickhouse-connect/default.nix:60]: Because I forget about all the ways there are to ignore tests 😂

**Comment by cpcloud** [comment]
Result of `nixpkgs-review pr 246170` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>14 packages built:</summary>
  <ul>
    <li>mycli</li>
    <li>mycli.dist</li>
    <li>python310Packages.clickhouse-connect</li>
    <li>python310Packages.clickhouse-connect.dist</li>
    <li>python310Packages.ibis-framework</li>
    <li>python310Packages.ibis-framework.dist</li>
    <li>python310Packages.sqlglot</li>
    <li>python310Packages.sqlglot.dist</li>
    <li>python311Packages.clickhouse-connect</li>
    <li>python311Packages.clickhouse-connect.dist</li>
    <li>python311Packages.ibis-framework</li>
    <li>python311Packages.ibis-framework.dist</li>
    <li>python311Packages.sqlglot</li>
    <li>python311Packages.sqlglot.dist</li>
  </ul>
</details>

**Pull request merged**